src/tools/diagnostics.md

```diff
@@ -0,0 +1,2436 @@
+
+#### Example
+
+The following code generates this diagnostic:
+
+{% prettify dart %}
+class Invalid extends [!Duration?!] {}
+{% endprettify %}
+
+#### Common fixes
+
+The most common fix is to remove the question mark:
+
+```dart
+class Invalid extends Duration {}
+```
+
+## Hints
+
+### DEPRECATED_MEMBER_USE
+
+#### Message
+
+'{0}' is deprecated and shouldn't be used.
+
+#### Description
+
+The analyzer produces this diagnostic when a deprecated library or class member is used in a different package.
+
+#### Example
+
+If the method `m` in the class `C` is annotated with `@deprecated`, then the following code produces this diagnostic:
+
+{% prettify dart %}
+void f(C c) {
+  c.[!m!]();
+}
+{% endprettify %}
+
+#### Common fixes
+
+The documentation for elements that are annotated with `@deprecated` should have documentation to indicate what code to use in place of the deprecated code.
+
+### INVALID_LITERAL_ANNOTATION
+
+#### Message
+
+Only const constructors can have the `@literal` annotation.
+
+#### Description
+
+This diagnostic occurs when the `@literal` annotation is on anything that isn't a const constructor. The annotation is defined to apply to const constructors, so its use anywhere else doesn't have a well defined meaning.
+
+The meaning of the `@literal` annotation is only defined when it's applied to a const constructor.
+
+#### Example
+
+The following code produces this diagnostic:
+
+{% prettify dart %}
+[!@literal!]
+var x;
+{% endprettify %}
+
+#### Common fixes
+
+Remove the annotation:
+
+```dart
+var x;
+```
+
+### SDK_VERSION_SET_LITERAL
+
+#### Message
+
+Set literals weren't supported until version 2.2, but this code must be able to run on earlier versions.
+
+#### Description
+
+The analyzer produces this diagnostic when a set literal is found in code that has an SDK constraint whose lower bound is less than 2.2. Set literals were not supported in earlier versions, so this code won't be able to run against earlier versions of the SDK.
+
+#### Example
+
+In a package that defines SDK constraints in the `pubspec.yaml` file that have a lower bound that's less than 2.2:
+
+```dart
+environment:
+  sdk: '>=2.1.0 <2.4.0'
+```
+
+The following code generates this diagnostic:
+
+{% prettify dart %}
+var s = [!<int>{}!];
+{% endprettify %}
+
+#### Common fixes
+
+If you don't need to support older versions of the SDK, then you can increase the SDK constraint to allow the syntax to be used:
+
+```dart
+environment:
+  sdk: '>=2.2.0 <2.4.0'
+```
+
+If you do need to support older versions of the SDK, then replace the set literal with code that creates the set without the use of a literal:
+
+```dart
+var s = new Set<int>();
+```
+
+{% comment %}
```

Comment:
I'm not planning to make the generator generate any of the content from this line down. Let me know if there's some value to doing so.